srtcore/api.cpp

```diff
@@ -929,7 +937,7 @@ int CUDTUnited::connect(SRTSOCKET u, const sockaddr* srcname, int srclen, const
 
     // For a single socket, just do bind, then connect
     bind(s, source_addr);
-    return connectIn(s, target_addr, 0);
+    return connectIn(s, target_addr, -1);
```

Comment:
-1 sets the initial sequence number to -1 instead of generating a random number. Was it the intent?

Reply:
The intent is to make this value as "no sequence number, please generate". The problem that I haven't seen for a long time after I introduced this feature was that I missed that 0 is actually a valid sequence number - tho probably prevented from being a result of random generation. The -1 value is the only "trap representation" for sequence number (a value that will never be a valid sequence number).

Reply:
Found it. Yes, there's a bug. `forced_isn` should be compared to -1.

Reply:
Deserves a separate PR BTW.

Reply:
Created PR #1099